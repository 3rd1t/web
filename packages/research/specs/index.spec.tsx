import React from "react"
import renderer from "react-test-renderer"
import Index from "../pages/index"

describe(" Index", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Index codeExamples={[{ language: "r", snippet: "r <- 1" }]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
