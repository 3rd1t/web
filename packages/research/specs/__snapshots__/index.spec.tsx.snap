// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Index with code snippets renders correctly 1`] = `
<body>
  <div>
    <div>
      <div
        class="pt-16 bg-gray-100"
      >
        <div
          class="fixed inset-x-0 top-0 z-20 bg-gray-100"
        >
          <nav
            class="container flex flex-wrap items-center justify-between h-16 px-4 mx-auto md:flex-no-wrap"
            role="navigation"
          >
            <a
              href="/"
            >
              <div
                class="flex items-center space-x-2"
              >
                <svg
                  class="h-6 text-gray-900 stroke-current"
                  fill="none"
                  viewBox="0 0 194 148"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 106.208L97.3883 17.8199L134.158 54.5894M185.07 41.8615L96.6814 130.25L59.9118 93.4803"
                    stroke-width="25"
                  />
                </svg>
                <div
                  class="text-xl font-semibold"
                >
                  <span
                    class="text-gray-800"
                  >
                    PERFOLIO
                  </span>
                  <span
                    class="uppercase text-research-500"
                  >
                    research
                  </span>
                </div>
              </div>
            </a>
            <div
              class="hidden md:block"
            >
              <div
                class="w-auto"
              >
                <ul
                  class="flex items-center space-x-10"
                >
                  <li>
                    <a
                      class="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
                      href="/"
                    >
                      Product
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
                      href="/#features"
                    >
                      Features
                    </a>
                  </li>
                  <li>
                    <a
                      class="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
                      href="/docs"
                    >
                      Documentation
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="block md:hidden"
            >
              <div>
                <button
                  aria-label="open menu"
                  class="p-2 text-gray-700 hover:text-gray-900 focus:outline-none"
                  data-testid="toggle"
                >
                  <svg
                    class="w-6 h-6"
                    fill="none"
                    stroke="currentColor"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 6h16M4 12h16M4 18h16"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </nav>
        </div>
        <section
          class="bg-gray-100 "
          id="index"
        >
          <div
            class="min-h-screen"
          >
            <div
              class="container px-4 py-10 mx-auto md:py-12 lg:py-16 xl:py-20 xl:px-0"
            >
              <div
                class="flex flex-col items-center px-4 space-y-8 sm:pt-24 md:pt-28 lg:pt-32 xl:pt-40 lg:flex-row lg:space-y-0 lg:space-x-8"
              >
                <div>
                  <div
                    class="flex flex-col items-center w-full"
                  >
                    <div
                      class="space-y-4 text-left sm:text-center md:space-y-8 xl:space-y-12 lg:text-left"
                    >
                      <h2
                        class="text-4xl font-extrabold leading-10 tracking-tight text-carbon-900 xl:text-5xl sm:leading-none md:text-5xl"
                      >
                        Get the right factor returns from our research-friendly API
                      </h2>
                      <div
                        class="text-carbon-600 sm:text-lg sm:mx-auto md:text-xl lg:mx-0"
                      >
                        <p
                          class="flex flex-col xl:flex-row"
                        >
                          We offer Kenneth French’s factor returns data via convenient REST API. No csv imports and data cleaning needed. Just go ahead with your analysis.
                        </p>
                      </div>
                      <div
                        class="space-y-4 md:space-x-4 md:space-y-0"
                      >
                        <div
                          class="sm:flex sm:justify-center lg:justify-start"
                        >
                          <div
                            class="rounded-md shadow"
                          >
                            <a
                              class="rounded-sm md:py-4 md:text-lg md:px-10 flex items-center justify-center w-full px-8 py-3 text-base font-medium leading-6 transition duration-200 ease-in-out focus:outline-none text-white bg-research-600 hover:bg-research-800"
                              href="#features"
                            >
                              Start now
                            </a>
                          </div>
                          <div
                            class="mt-3 sm:mt-0 sm:ml-3"
                          >
                            <a
                              class="rounded-sm md:py-4 md:text-lg md:px-10 flex items-center justify-center w-full px-8 py-3 text-base font-medium leading-6 transition duration-200 ease-in-out focus:outline-none text-research-900 hover:text-research-700"
                              href="mailto:info@perfol.io"
                            >
                              Contact us
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="hidden lg:block"
                >
                  <div
                    class="shadow-lg"
                  >
                    <nav
                      class="flex items-center justify-start px-4 space-x-4 border border-gray-500 rounded-t"
                    >
                      <button
                        class="py-3 mr-8 text-xs font-semibold tracking-wide no-underline uppercase focus:outline-none hover:text-gray-900 text-gray-900 "
                        data-testid="selectLanguage"
                      >
                        r
                      </button>
                    </nav>
                    <div
                      class="px-4 overflow-x-visible overflow-y-hidden bg-gray-900 rounded-b"
                    >
                      <pre
                        class="font-mono"
                        data-testid="syntaxHighlighter"
                        style="font-size: 14px; line-height: 1.375; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; tab-size: 4; hyphens: none; color: rgb(203, 213, 224); padding: 1em; margin: .5em 0px; overflow: auto;"
                      >
                        <code
                          class="language-r"
                          style="font-size: 14px; line-height: 1.375; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; tab-size: 4; hyphens: none; color: rgb(203, 213, 224);"
                        >
                          r &lt;- 1
                        </code>
                      </pre>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="bg-white"
          id="features"
        >
          <div>
            <div
              class="container px-4 py-10 mx-auto md:py-12 lg:py-16 xl:py-20 xl:px-0"
            >
              <div
                class=" lg:text-center"
              >
                <h3
                  class="mt-2 text-3xl font-extrabold tracking-tight text-gray-900 sm:text-4xl sm:leading-10"
                >
                  Kenneth French’s factor returns the way you want
                </h3>
                <p
                  class="mt-3 text-lg text-gray-700"
                >
                  Conversion to your home currency included.
                </p>
              </div>
              <div
                class="flex flex-col justify-center mt-20 space-y-4 lg:space-x-4 lg:space-y-0 lg:flex-row"
              >
                <div
                  class="flex flex-col justify-between -space-y-2 overflow-hidden duration-100 transform bg-white border rounded-md lg:hover:scale-105 lg:w-1/3 hover:shadow-xl hover:border-2 border-research-500"
                >
                  <h1
                    class="p-4 text-4xl font-black text-center lg:text-3xl md:text-lg xl:text-5xl text-carbon-900"
                  >
                    REST API
                  </h1>
                  <div
                    class="flex flex-col items-center justify-center p-4 bg-gray-100 lg:items-start"
                  >
                    <ul
                      class="mt-6 space-y-3 text-gray-800"
                    >
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Any programing language
                        </span>
                      </li>
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Always up to date
                        </span>
                      </li>
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Code examples available
                        </span>
                      </li>
                    </ul>
                    <a
                      class="w-full px-4 py-2 mt-6 text-sm font-semibold text-center text-gray-100 bg-gray-900 rounded focus:outline-none hover:bg-research-800"
                      href="/api"
                    >
                      Get started
                    </a>
                  </div>
                </div>
                <div
                  class="flex flex-col justify-between -space-y-2 overflow-hidden duration-100 transform bg-white border rounded-md lg:hover:scale-105 lg:w-1/3 hover:shadow-xl hover:border-2 border-research-500"
                >
                  <h1
                    class="p-4 text-4xl font-black text-center lg:text-3xl md:text-lg xl:text-5xl text-carbon-900"
                  >
                    CSV
                  </h1>
                  <div
                    class="flex flex-col items-center justify-center p-4 bg-gray-100 lg:items-start"
                  >
                    <ul
                      class="mt-6 space-y-3 text-gray-800"
                    >
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Use directly in Excel
                        </span>
                      </li>
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Straightforward information schema
                        </span>
                      </li>
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Easy to handle
                        </span>
                      </li>
                    </ul>
                    <a
                      class="w-full px-4 py-2 mt-6 text-sm font-semibold text-center text-gray-100 bg-gray-900 rounded focus:outline-none hover:bg-research-800"
                      href="/csv"
                    >
                      Get started
                    </a>
                  </div>
                </div>
                <div
                  class="flex flex-col justify-between -space-y-2 overflow-hidden duration-100 transform bg-white border rounded-md lg:hover:scale-105 lg:w-1/3 hover:shadow-xl hover:border-2 border-research-500"
                >
                  <h1
                    class="p-4 text-4xl font-black text-center lg:text-3xl md:text-lg xl:text-5xl text-carbon-900"
                  >
                    Charts
                  </h1>
                  <div
                    class="flex flex-col items-center justify-center p-4 bg-gray-100 lg:items-start"
                  >
                    <ul
                      class="mt-6 space-y-3 text-gray-800"
                    >
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Analyse directly in your browser
                        </span>
                      </li>
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Who doesn't like charts
                        </span>
                      </li>
                      <li
                        class="flex items-center space-x-3"
                      >
                        <svg
                          class="w-5 h-5 text-research-500"
                          fill="currentColor"
                          viewBox="0 0 20 20"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            fill-rule="evenodd"
                          />
                        </svg>
                        <span>
                          Visual reprerentation
                        </span>
                      </li>
                    </ul>
                    <a
                      class="w-full px-4 py-2 mt-6 text-sm font-semibold text-center text-gray-100 bg-gray-900 rounded focus:outline-none hover:bg-research-800"
                      href="/charts"
                    >
                      Get started
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="bg-gray-100"
          id="charts"
        >
          <div>
            <div
              class="container px-4 py-10 mx-auto md:py-12 lg:py-16 xl:py-20 xl:px-0"
            >
              <div
                class=" lg:text-center"
              >
                <h3
                  class="mt-2 text-3xl font-extrabold tracking-tight text-gray-900 sm:text-4xl sm:leading-10"
                >
                  Charts
                </h3>
                <p
                  class="mt-3 text-lg text-gray-700"
                />
              </div>
              <div
                class="flex justify-center mt-16"
              >
                <img
                  src="https://via.placeholder.com/700x300"
                />
              </div>
            </div>
          </div>
        </section>
        <section
          class="bg-white"
          id="sources"
        >
          <div>
            <div
              class="container px-4 py-10 mx-auto md:py-12 lg:py-16 xl:py-20 xl:px-0"
            >
              <div
                class=" lg:text-center"
              >
                <h3
                  class="mt-2 text-3xl font-extrabold tracking-tight text-gray-900 sm:text-4xl sm:leading-10"
                >
                  Sources
                </h3>
                <p
                  class="mt-3 text-lg text-gray-700"
                >
                  We apply the latest finance research findings. Please give the authors and us some credit and cite correctly.
                </p>
              </div>
              <div
                class="px-4 mt-20 divide-y divide-gray-400"
              >
                <div>
                  <div
                    class="flex flex-wrap py-8 md:flex-no-wrap"
                  >
                    <div
                      class="flex flex-col flex-shrink-0 pr-2 mb-6 md:w-64 md:mb-0"
                    >
                      <span
                        class="font-medium tracking-widest text-gray-900"
                      >
                        Kenneth R. French
                      </span>
                      <span
                        class="mt-1 text-sm text-gray-600"
                      >
                        06.08.2020
                      </span>
                    </div>
                    <div
                      class="space-y-4 md:flex-grow"
                    >
                      <h2
                        class="text-2xl text-gray-900"
                      >
                        Kenneth French's Data library
                      </h2>
                      <p
                        class="text-gray-700"
                      >
                        We have permission to distribute his library blabla, maybe something like this?
                      </p>
                      <div
                        class="flex items-center space-x-4"
                      >
                        <a
                          class="text-xs text-gray-900 focus:outline-none hover:text-research-600"
                          href="http://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html"
                          target="blank"
                        >
                          Go to source
                        </a>
                        <button
                          class="text-xs text-gray-900 focus:outline-none hover:text-research-600"
                        >
                          Citation
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section>
          <div
            class="bg-gray-100"
          >
            <div
              class="container px-4 py-10 mx-auto md:py-12 lg:py-16 xl:py-20 xl:px-0"
            >
              <div
                class="flex flex-col items-center justify-between space-y-4 md:flex-row md:space-y-0"
              >
                <div>
                  <h2
                    class="text-3xl font-extrabold leading-9 tracking-tight text-gray-900 sm:text-4xl sm:leading-10"
                  >
                    <span>
                      Ready to get started? 
                    </span>
                    <br
                      class="xl:hidden"
                    />
                    <span>
                      Start with a
                    </span>
                    <span
                      class="text-research-500"
                    >
                       free plan
                    </span>
                  </h2>
                  <p
                    class="mt-2 text-gray-700"
                  >
                    Explore our charts or create an account to unlock the full potential of our API.
                  </p>
                </div>
                <a
                  class="block w-full px-6 py-3 font-semibold text-white transition duration-150 ease-in-out border border-transparent rounded shadow-md bg-carbon-900 sm:mt-0 sm:h-auto sm:ml-4 sm:w-auto hover:bg-carbon-800 focus:outline-none focus:bg-carbon-800 hover:bg-carbon-600"
                  href="/signup"
                >
                  Sign up
                </a>
              </div>
            </div>
          </div>
        </section>
      </div>
      <footer
        class="bg-research-700"
      >
        <div
          class="container mx-auto"
        >
          <div
            class="flex flex-wrap justify-between p-10  text-gray-100"
          >
            <div
              class="p-2 md:space-y-2"
            >
              <span
                class="text-xs font-semibold text-white uppercase"
              >
                Projects
              </span>
              <a
                class="block focus:outline-none text-gray-100 hover:text-white"
                href="https://perfol.io"
              >
                Home
              </a>
              <div
                class="flex items-center space-x-2"
              >
                <a
                  class="block focus:outline-none text-gray-100 hover:text-white"
                  href="https://analytics.perfol.io"
                >
                  Analytics
                </a>
                <span
                  class="px-1 text-sm bg-purple-800 rounded-sm"
                >
                  WIP
                </span>
              </div>
              <a
                class="block focus:outline-none text-gray-100 hover:text-white"
                href="https://research.perfol.io"
              >
                Research
              </a>
            </div>
            <div
              class="p-2 md:space-y-2"
            >
              <span
                class="text-xs font-semibold text-white uppercase"
              >
                Resources
              </span>
              <a
                class="block focus:outline-none text-gray-100 hover:text-white"
                href="/docs"
              >
                Documentation
              </a>
              <a
                class="block focus:outline-none text-gray-100 hover:text-white"
                href="/imprint"
              >
                Imprint
              </a>
              <a
                class="block focus:outline-none text-gray-100 hover:text-white"
                href="/privacy"
              >
                Privacy
              </a>
            </div>
            <div
              class="p-2 md:space-y-2"
            >
              <span
                class="text-xs font-semibold text-white uppercase"
              >
                Contact us
              </span>
              <a
                class="block focus:outline-none text-gray-100 hover:text-white"
                href="mailto:info@perfol.io"
              >
                contact@perfol.io
              </a>
            </div>
            <div
              class="p-2 md:space-y-2"
            >
              <span
                class="text-xs font-semibold text-white uppercase"
              >
                Source
              </span>
              <a
                class="block focus:outline-none text-gray-100 hover:text-white"
                href="https://github.com/perfolio"
              >
                github.com
              </a>
            </div>
          </div>
        </div>
        <div
          class="pt-2 border-t border-research-500"
        >
          <div
            class="max-w-6xl mx-auto"
          >
            <div
              class="flex flex-col items-center px-10 py-2 text-sm md:flex-row"
            >
              <span
                class="text-white"
              >
                © Copyright 
                2020
                . All Rights Reserved.
              </span>
              <div
                class="flex flex-row md:flex-auto md:flex-row-reverse"
              >
                <a
                  href="https://www.netlify.com"
                >
                  <img
                    alt="Deploys by Netlify"
                    src="https://www.netlify.com/img/global/badges/netlify-light.svg"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </div>
</body>
`;
